lib/Macros/Sources/ObservationMacros/ObservableMacro.swift

```diff
@@ -180,8 +187,57 @@ extension TokenSyntax {
   }
 }
 
+extension CodeBlockSyntax {
+  func locationAnnotated(in context: LocalMacroExpansionContext<some MacroExpansionContext>) -> CodeBlockSyntax {
+    guard let converter = context.converter, let firstStatement = statements.first else {
+      return self
+    }
+    let range = firstStatement.sourceRange(converter: converter)
+    return CodeBlockSyntax(
+      leadingTrivia: leadingTrivia,
+      leftBrace: leftBrace,
+      statements: CodeBlockItemListSyntax {
+        "#sourceLocation(file: \(context.file), line: \(raw: range.start.line))"
+        statements
+        "#sourceLocation()"
+      },
+      rightBrace: rightBrace,
+      trailingTrivia: trailingTrivia
+    )
+  }
+}
+
+extension AccessorDeclSyntax {
+  func locationAnnotated(in context: LocalMacroExpansionContext<some MacroExpansionContext>) -> AccessorDeclSyntax {
+    return AccessorDeclSyntax(
+      leadingTrivia: leadingTrivia,
+      attributes: attributes,
+      modifier: modifier,
+      accessorSpecifier: accessorSpecifier,
+      parameters: parameters,
+      effectSpecifiers: effectSpecifiers,
+      body: body?.locationAnnotated(in: context),
+      trailingTrivia: trailingTrivia
+    )
+  }
+}
+
+extension AccessorBlockSyntax {
+  func locationAnnotated(in context: LocalMacroExpansionContext<some MacroExpansionContext>) -> AccessorBlockSyntax {
+    switch accessors {
+    case .accessors(let accessorList):
+      let remapped = AccessorDeclListSyntax {
+        accessorList.map { $0.locationAnnotated(in: context) }
+      }
+      return AccessorBlockSyntax(accessors: .accessors(remapped))
+    case .getter(let codeBlockList):
+      return AccessorBlockSyntax(accessors: .getter(codeBlockList))
```

Comment:
Also needs remapping?

Reply:
Im actually not sure this can even occur since the properties applied here are only stored; ones with getters are invalid for synthesis like this (so it could be a fatal error technically... but id rather leave the existing behavior there than add a bomb)